File: framework/ios/Example/Example/OpenpathWrapper/Models/AuthorizationStatus.swift
import OpenpathMobile

extension AuthorizationStatusType: Decodable {}

enum AuthType: String, CaseIterable, Decodable {
    case location
    case motion
    case bluetooth
    case notification
    case microphone
}

struct AuthorizationStatus: Decodable {
    let authType: AuthType
    let status: AuthorizationStatusType
}

File: types/Models/AuthorizationStatus.ts
export type AuthType =
  | 'location'
  | 'motion'
  | 'bluetooth'
  | 'notification'
  | 'microphone';

export enum AuthorizationStatusType {
  notDetermined = -1,
  denied,
  granted,
}

export interface AuthorizationStatus {
  authType: AuthType;
  status: AuthorizationStatusType;
}
